### ISSUE TITLE
Master memory usage

### ISSUE THREAD

**Issue by guestisp** [lizardfs/lizardfs#543]
Any "rule of thumbs" about master memory usage? What if I need more RAM than available ?

In example, by using a server with 128GB, how many files/chunks/terabyte i'm able to manage and what If i need more RAM without being able to add (maybe server with limited amount or memory slots)

Can we safely swap on SSD? Is possible to force master to use SSD storage without keeping the whole metadata archive in RAM ?

**Comment by jkiebzak** [comment]
There was a partial discussion of this in #421. It would be nice to have an
on-disk metadata option, rather than being completely dependent of the
amount of RAM that one CPU can see on your box. Or, to be able to leverage
a dispersed key/value store of any sort.

**Comment by guestisp** [comment]
Exactly.... As master doesn't have benefits from multiple cpus, a single CPU is also limiting the ram available on the server

**Comment by aki009** [comment]
This paper (http://students.mimuw.edu.pl/~ps321169/mgr.pdf) seems to say that the data structure for a file takes about 500 bytes, including hashing etc. YMMV.
